lib/arena.rb

```diff
@@ -1,2 +1,135 @@
+# require("pry")
+require_relative "gladiator.rb"
+
 class Arena
+    attr_accessor :name, :gladiators
+
+    def initialize name                     # create a new Arena
+        @name = name.capitalize             # capitalize what the user enters
+        @gladiators = []                    # define a gladiators array that starts empty
+    end
+
+    def add_gladiator instance              # ASSUMING the user has already run Gladiator.new methods...
+        if @gladiators.length < 2           # ONLY if there are fewer than two gladiators already in the arena
+            @gladiators << instance         # add the gladiator instance to the gladiators array
+        end
+    end
+
+    def remove_gladiator instance
+        @gladiators.delete(instance)        # allow the user to remove gladiators from the arena
+    end
+
+    def entertained?
+        puts "Maximus: 'ARE YOU NOT ENTERTAINED?!?'"   # during the fight, ask if the crowd is entertained
+    end
+
+    ####################### FOR THE THUMBS UP/DOWN BONUS #######################
+    def thumbs_up_or_down winner, loser
+        puts `clear`                                                                          # clear the screen
+        puts "People of the #{@name}, should #{loser.name} die? Thumps 'up' or 'down':"       # for the bonus...
+        thumb = gets.to_s.chomp                                                               # ask the crowd for a
+        while thumb != "up" && thumb != "down"                                                # thumbs up or thumbs down
+            puts "Choose 'up' or 'down' or face the Emperor's wrath!"                         # force a readable response
+            thumb = gets.to_s.chomp
+        end
+        binding.pry
+        if thumb == "up"                                  # if it's a thumbs up...
+            puts "FINISH HIM!"                            # the crowd calls on the winner to finish the loser
+            5.times do
+                puts "."
+                sleep 0.2                                 # delay
+            end
+            puts "#{loser.name} is vanquished! Long live #{winner.name}!"    # the winner stays the winner
+            @gladiators.delete(loser)                                        # the loser dies
+            sleep 1.5
+            return winner
+        elsif thumb == "down" && winner.name == "Maximus"       # if it's thumbs down but Maximus is the winner...
+            puts "MAXIMUS!!! MAXIMUS THE MERCIFUL!!!"           # both gladiators survive!
+            5.times do
+                puts "."
+                sleep 0.2                                 # delay
+            end
+            puts "Both #{winner.name} and #{loser.name} will live to fight another day!"
+            sleep 1.5
```

Comment:
Might be an opportunity to refactor. Notice some of the code in these two conditionals is pretty similar.